Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -673,55 +725,112 @@ extension ChannelPipeline.SynchronousOperations {
+}
+
+/// `HTTP2AsyncConfiguration` contains all configuration required for setting up an HTTP/2 async connection.
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+@_spi(AsyncChannel)
+public struct HTTP2AsyncConfiguration<HTTP2ConnectionInbound, HTTP2ConnectionOutbound, HTTP2StreamInbound, HTTP2StreamOutbound> {
+    /// The settings that will be used when establishing new streams. These mainly pertain to flow control.
+    public var connection: NIOHTTP2Handler.ConnectionConfiguration
+    /// The settings that will be used when establishing the connection. These will be sent to the peer as part of the
+    /// handshake.
+    public var stream: NIOHTTP2Handler.StreamConfiguration
+
+    /// The settings for the `NIOAsyncChannel` wrapping the HTTP/2 connection channel
+    public var connectionAsyncChannel: NIOAsyncChannel<HTTP2ConnectionInbound, HTTP2ConnectionOutbound>.Configuration
+    /// The settings for `NIOAsyncChannel`s wrapping inbound HTTP/2 stream channels
+    public var inboundStreamAsyncChannel: NIOAsyncChannel<HTTP2StreamInbound, HTTP2StreamOutbound>.Configuration
+
+    /// A callback used to configure the HTTP/2 connection channel
+    public var connectionInitializer: NIOChannelInitializer
+    /// A callback used to configure inbound HTTP/2 stream channels
+    public var inboundStreamInitializer: NIOChannelInitializer
+
+    public init(
+        connectionAsyncChannel: NIOAsyncChannel<HTTP2ConnectionInbound, HTTP2ConnectionOutbound>.Configuration,
+        inboundStreamAsyncChannel: NIOAsyncChannel<HTTP2StreamInbound, HTTP2StreamOutbound>.Configuration,
+        connection: NIOHTTP2Handler.ConnectionConfiguration = .init(),
+        stream: NIOHTTP2Handler.StreamConfiguration = .init(),
+        connectionInitializer: @escaping NIOChannelInitializer,
+        inboundStreamInitializer: @escaping NIOChannelInitializer
+    ) {
+        self.connectionAsyncChannel = connectionAsyncChannel
+        self.inboundStreamAsyncChannel = inboundStreamAsyncChannel
+        self.connection = connection
+        self.stream = stream
+        self.connectionInitializer = connectionInitializer
+        self.inboundStreamInitializer = inboundStreamInitializer
+    }
+}
+
+
+// TODO: remove. This is a placeholder for the upcoming native type
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension NIOAsyncChannel {
+    @_spi(AsyncChannel)
+    public struct Configuration: Sendable{
+        let inboundType: Inbound.Type
+        let outboundType: Outbound.Type
+        let backpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark?
+        let isOutboundHalfClosureEnabled: Bool
+
+        public init(
+            inboundType: Inbound.Type = Inbound.self,
+            outboundType: Outbound.Type = Outbound.self,
+            backpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
+            isOutboundHalfClosureEnabled: Bool? = nil
```

Comment:
I think these were added in a different order in NIO: `inboundType` and `outboundType` came at the end